File: src_local/agents/Flutter/Audit/bolt.py
from src_local.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class BoltPersona(BaseActivePersona):
    """
    Core: PhD in Computational Efficiency (Flutter) ⚡
    Especialista em detecção de frames perdidos e loops bloqueantes em Dart.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Bolt", "⚡", "PhD Performance Engineer", "Flutter"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Eficiência no ecossistema Flutter...")
        
        # Regras específicas para Dart/Flutter
        audit_rules = [
            {'regex': r"while\s*\(true\)\s*\{\s*\}", 'issue': 'Gargalo: Busy-waiting detectado em Dart.', 'severity': 'critical'},
            {'regex': r"sleep\(", 'issue': 'Risco: Uso de sleep bloqueia a UI Thread do Flutter.', 'severity': 'high'},
            {'regex': r"for\s*\(var\s+i\s*=\s*0;\s*i\s*<\s*.*\.length;\s*i\+\+\)", 'issue': 'Otimização: Prefira .map() ou .forEach() para iteráveis em Dart.', 'severity': 'low'}
        ]
        
        results = self.find_patterns(('.dart',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        if "sleep(" in content:
            return f"Degradação de UX: O objetivo '{objective}' exige fluidez. Em '{file}', o uso de sleep() trava a Main Thread, impedindo a 'Orquestração de Inteligência Artificial' de manter 60fps."
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em performance Flutter. Sua missão é garantir zero jank."

File: src_local/agents/Flutter/Audit/bolt.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditRule, StrategicFinding, AuditFinding } from "../../base.ts";
import type { ProjectContext } from "../../../core/types.ts";

/**
 * 🏎️ Dr. Bolt — PhD in Flutter Computational Efficiency
 */
export class BoltPersona extends BaseActivePersona {
    constructor(projectRoot: string | undefined = undefined) {
        super(projectRoot);
        this.name = "Bolt";
        this.emoji = "⚡";
        this.role = "Sovereign Performance Architect";
        this.phd_identity = "Computational Efficiency & Runtime Optimization";
        this.stack = "Flutter";
    }

    override async execute(context: ProjectContext): Promise<AuditFinding[]> {
        this.setContext(context);
        const findings = await this.performAudit();
        return findings;
    }

    getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.dart'],
            rules: [
                { regex: /while\s*\(true\)\s*\{/, issue: 'Gargalo: Loop infinito detectado em Dart.', severity: 'critical' }
            ]
        };
    }

    public audit(): any[] { return []; }
    public Branding(): string { return `${this.emoji} ${this.name}`; }
    public Analysis(): string { return "Performance Analysis Complete"; }

    public test(): boolean {
        this.audit();
        this.Branding();
        this.Analysis();
        return true;
    }

    override reasonAboutObjective(objective: string, file: string, _content: string | Promise<string | null>): StrategicFinding | string | null {
        return {
            file, severity: "INFO",
            issue: `PhD Bolt (Flutter): Analisando eficiência para ${objective}.`,
            context: "analyzing efficiency"
        };
    }

    override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [], branding: this.Branding() };
    }

    override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, mestre em performance Flutter. Status: ${this.Analysis()}`;
    }
}
